dir/path.go

```diff
@@ -79,21 +79,35 @@ const (
 
 var userConfigDir = os.UserConfigDir // for unit test
 
-func init() {
-	loadUserPath()
-}
-
 // loadUserPath function defines UserConfigDir and UserLibexecDir.
 func loadUserPath() {
-	// set user config
-	userDir, err := userConfigDir()
-	if err != nil {
-		panic(err)
+	UserConfigDir = ""
+	UserLibexecDir = ""
+
+	_ = UserLibexecDirPath()
+}
+
+// UserConfigDirPath returns the user level {NOTATION_CONFIG} path.
+func UserConfigDirPath() string {
+	if UserConfigDir == "" {
+		userDir, err := userConfigDir()
+		if err != nil {
+			// fallback to current directory
+			userDir = "."
```

Comment:
By including this PR and updating the Notation CLI, if the `$HOME` and `$XDG_CONFIG_HOME` variables are not set on Linux, we will use the current working directory as the default config directory. This is different from the previous implementation, where the application would panic in this case.

Comment for PM @yizha1 @FeynmanZhou

Reply:
We need to update this document https://notaryproject.dev/docs/user-guides/how-to/directory-structure/ to show how the config directory is determined.

Reply:
@JeyJeyGao see https://github.com/notaryproject/notation-go/pull/449#discussion_r1741356288. In summary, rather than `panic`, if `$HOME` and `$XDG_CONFIG_HOME` are not set, `./.notation` is used instead. Otherwise, `$HOME/.config/notation` is used.

Reply:
I created a new issue to track the documentation change requirements https://github.com/notaryproject/notaryproject.dev/issues/413